### PR TITLE
Fixed fatal logs

### PR DESCRIPTION
**Pull request by Missing-Parallax**

### CODE REVIEW

File: utils/kafka.go [hunk 1]
```diff
@@ -14,20 +14,20 @@ func PublishMessage(kafkaURI, topic string, message string) string {
 
 	conn, err := kafka.DialLeader(context.Background(), "tcp", kafkaURI, topic, partition)
 	if err != nil {
-		log.Fatal("failed to dial leader:", err)
+		log.Println("failed to dial leader:", err)
```
Comment (uzaxirr) [inline utils/kafka.go:17]: I would recommend to use `log.Error()` here

@abhishekraj272 Do we want to terminate this function in case error arises here?

**Review by uzaxirr — CHANGES_REQUESTED** [review]

File: utils/kafka.go (hunk 1 above)
Comment (abskrj) [reply utils/kafka.go:17]: yeah, returning error would be better approach

Comment (abskrj) [reply utils/kafka.go:17]: @uzaxirr you mean return err? there is no function as log.Error

Comment (uzaxirr) [reply utils/kafka.go:17]: > @uzaxirr you mean return err? there is no function as log.Error


Any Error Level log is fine here, and then return an error from the function

Comment (uzaxirr) [reply utils/kafka.go:17]: @abhishekraj272 
Do we also need to do error handling [here](https://github.com/ZeStream/zestream-server/blob/dev/controllers/kafka.go#L33) if this function returns an error, Ideally this Publish function should return a `message` and an `error`, and in the controller we should have checks for this error.

What do you think?

Comment (abskrj) [reply utils/kafka.go:17]: this function should return the error and the response should be sent from the controller

Comment (uzaxirr) [reply utils/kafka.go:17]: @Missing-Parallax 
Do you wish to incorporate these changes in this PR itself or should i create a new issue for it?

**Comment by uzaxirr** [comment]
Hey @Missing-Parallax 
Any updates?

**Comment by Missing-Parallax** [comment]
> Hey @Missing-Parallax Any updates?

Sorry, I am having exam tomorrow, so I did not check.
Will check after the exam, if thts not an issue

**Comment by uzaxirr** [comment]
> > Hey @Missing-Parallax Any updates?
> 
> Sorry, I am having exam tomorrow, so I did not check. Will check after the exam, if thts not an issue

Not a problem at all

**Comment by uzaxirr** [comment]
Closing this wrt #56

**Pull request closed without merging**